R/groups.R

```diff
@@ -0,0 +1,190 @@
+#'
+#' @details
+#' Please see https://docs.posit.co/connect/api/#get-/v1/groups/-group_guid-/members
+#' for more information.
+#'
+#' @examples
+#' \dontrun{
+#' library(connectapi)
+#' client <- connect()
+#'
+#' # get the first 20 groups
+#' groups <- get_groups(client)
+#'
+#' group_guid <- groups$guid[1]
+#'
+#' get_group_members(client, guid = group_guid)
+#' }
+#'
+#' @family groups functions
+#' @export
+get_group_members <- function(src, guid) {
+  validate_R6_class(src, "Connect")
+
+  res <- src$group_members(guid)
+
+  out <- parse_connectapi(res$results)
+
+  return(out)
+}
+
+#' Get content access permissions for a group or groups
+#'
+#' @param src A Connect client object
+#' @param guids A character vector of group guids
+#'
+#' @return
+#' A tibble with the following columns:
+
+#'   * `group_guid`: The group's GUID
+#'   * `group_name`: The group's name
+#'   * `content_guid`: The content item's GUID
+#'   * `content_name`: The content item's name
+#'   * `content_title`: The content item's title
+#'   * `access_type`: The access type of the content item ("all", "logged_in", or "acl")
+#'   * `role`: The access type that members of the group have to the
+#'     content item, "publisher" or "viewer".
+#'
+#' @examples
+#' \dontrun{
+#' library(connectapi)
+#' client <- connect()
+#'
+#' # Get a data frame of groups
+#' groups <- get_groups(client)
+#'
+#' # Get permissions for a single group by passing in the corresponding row.
+#' get_group_content(client, groups[1, "guid"])
+#' dplyr::filter(groups, name = "research_scientists") %>%
+#'   dplyr::pull(guid) %>%
+#'   get_group_content(client, .)
+#'
+#' # Get permissions for all groups by passing in all group guids.
+#' get_group_content(client, groups$guid)
+#' }
+#'
+#' @family groups functions
+#' @export
+#' @importFrom rlang .data
+get_group_content <- function(src, guids) {
+  validate_R6_class(src, "Connect")
+
+  purrr::map_dfr(
+    guids,
+    ~ get_group_content_impl(src = src, guid = .x)
+  )
+}
+
+#' @importFrom rlang .data
+get_group_content_impl <- function(src, guid) {
+  validate_R6_class(src, "Connect")
+
+  res <- src$group_content(guid)
+  if (length(res) == 0) {
+    return(tibble::tibble(
+      group_guid = NA_character_,
+      group_name = NA_character_,
+      content_guid = NA_character_,
+      content_name = NA_character_,
+      content_title = NA_character_,
+      access_type = NA_character_,
+      role = NA_character_
+    ))
+  }
+  parsed <- parse_connectapi_typed(res, connectapi_ptypes$group_content)
+
+  permissions_df <- purrr::map_dfr(
+    parsed$permissions,
+    ~ purrr::keep(
+      .x,
+      ~ .x[["principal_guid"]] == guid
+    )
+  )
+
+  dplyr::transmute(parsed,
+    group_guid = guid,
+    group_name = permissions_df$principal_name,
+    .data$content_guid,
+    .data$content_name,
+    .data$content_title,
+    .data$access_type,
+    role = permissions_df$principal_role
+  )
```

Comment:
```suggestion
    tibble::tibble(
    group_guid = guid,
    group_name = permissions_df$principal_name,
    content_guid = parsed$content_guid,
    content_name = parsed$content_name,
    content_title = parsed$content_title,
    access_type = parsed$access_type,
    role = permissions_df$principal_role
  )
```

What about this? it avoids a call to dplyr entirely (and transmute is superseded anyway)

Reply:
Lol yes `transmute` is totally irrelevant here good catch 😂